src/main/java/org/jabref/gui/search/RebuildFulltextSearchIndexAction.java

```diff
@@ -0,0 +1,75 @@
+package org.jabref.gui.search;
+
+import java.io.IOException;
+
+import org.jabref.gui.DialogService;
+import org.jabref.gui.Globals;
+import org.jabref.gui.LibraryTab;
+import org.jabref.gui.StateManager;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.util.BackgroundTask;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.pdf.search.indexing.PdfIndexer;
+import org.jabref.model.database.BibDatabaseContext;
+import org.jabref.preferences.FilePreferences;
+
+import static org.jabref.gui.actions.ActionHelper.needsDatabase;
+
+public class RebuildFulltextSearchIndexAction extends SimpleCommand {
+
+    private final StateManager stateManager;
+    private final GetCurrentLibraryTab currentLibraryTab;
+    private final DialogService dialogService;
+    private final FilePreferences filePreferences;
+
+    private BibDatabaseContext databaseContext;
+
+    private boolean shouldContinue = true;
+
+    public RebuildFulltextSearchIndexAction(StateManager stateManager, GetCurrentLibraryTab currentLibraryTab, DialogService dialogService, FilePreferences filePreferences) {
+        this.stateManager = stateManager;
+        this.currentLibraryTab = currentLibraryTab;
+        this.dialogService = dialogService;
+        this.filePreferences = filePreferences;
+
+        this.executable.bind(needsDatabase(stateManager));
+    }
+
+    @Override
+    public void execute() {
+        init();
+        BackgroundTask.wrap(this::rebuildIndex)
+                      .executeWith(Globals.TASK_EXECUTOR);
+    }
+
+    public void init() {
+        if (stateManager.getActiveDatabase().isEmpty()) {
+            return;
+        }
+
+        databaseContext = stateManager.getActiveDatabase().get();
+        boolean confirm = dialogService.showConfirmationDialogAndWait(
+                Localization.lang("Rebuild fulltext search index"),
+                Localization.lang("Rebuild fulltext search index for current library?"));
+        if (!confirm) {
+            shouldContinue = false;
+            return;
+        }
+        dialogService.notify(Localization.lang("Rebuilding fulltext search index..."));
+    }
+
+    private void rebuildIndex() {
+        if (!shouldContinue || stateManager.getActiveDatabase().isEmpty()) {
+            return;
+        }
+        try {
+            currentLibraryTab.get().getIndexingTaskManager().createIndex(PdfIndexer.of(databaseContext, filePreferences), databaseContext.getDatabase(), databaseContext);
+        } catch (IOException e) {
+            dialogService.notify(Localization.lang("Failed to access fulltext search index"));
```

Comment:
I would also log the original exception, as an IOException can have many reasons it's otherwise hard to debug.
And then  dialogService.notify(Localization.lang("Failed to access fulltext search index." See error log for details.));